src/cluster.c

```diff
@@ -3110,8 +3152,13 @@ void clusterSendPing(clusterLink *link, int type) {
     hdr->count = htons(gossipcount);
     hdr->extensions = htons(extensions);
     hdr->totlen = htonl(totlen);
-    clusterSendMessage(link,buf,totlen);
-    zfree(buf);
+
+    /* Resize message block to actual size */
+    clusterMsgSendBlock *resized_msgblock = createClusterMsgSendBlock(totlen);
```

Comment:
Do we need to create a new msgblock? Could we update the msg of the existing variable ?

Reply:
We could but I was noticing that `estlen` is significantly bigger than the resulting `totlen` in a small cluster, so I figured freeing memory sooner than later was a good call.

Example byte lengths when I launch a three node cluster:
```
estlen=4352
totlen=2360
```